contracts/staking/ServiceStakingBase.sol

```diff
@@ -541,38 +593,78 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
-        ServiceInfo memory sInfo = mapServiceInfo[serviceId];
-        reward = sInfo.reward;
+    /// @dev Evicts the service due to its extended inactivity.
+    function evict(uint256 serviceId) external {
+        ServiceInfo storage sInfo = mapServiceInfo[serviceId];
+        // Get the service inactivity time
+        uint256 inactivity = sInfo.inactivity;
+
+        // Evict the service if it is inactive more than the max number of inactivity periods
+        if (inactivity > maxAllowedInactivity) {
+            // Get service Ids to find the service
+            uint256[] memory serviceIds = getServiceIds();
+
+            // Get the service index in the set of services
+            // The index must always exist if the service is currently staked, otherwise it never existed or was evicted
+            uint256 idx;
+            for (; idx < serviceIds.length; ++idx) {
+                if (serviceIds[idx] == serviceId) {
+                    break;
+                }
+            }
 
-        // Check if the service is staked
-        if (sInfo.tsStart == 0) {
-            revert ServiceNotStaked(serviceId);
+            // The service was already evicted
+            if (idx == serviceIds.length) {
+                revert ServiceNotFound(serviceId);
+            }
+
+            // Evict the service from the set of staked services
+            setServiceIds[idx] = setServiceIds[setServiceIds.length - 1];
+            setServiceIds.pop();
+
+            emit ServiceEvicted(serviceId, sInfo.owner, sInfo.multisig, inactivity);
         }
+    }
 
+    /// @dev Calculates service staking reward during the last checkpoint period.
+    /// @param serviceId Service Id.
+    /// @return reward Service reward.
+    function calculateServiceStakingLastReward(uint256 serviceId) public view returns (uint256 reward) {
         // Calculate overall staking rewards
         (uint256 lastAvailableRewards, uint256 numServices, uint256 totalRewards, uint256[] memory eligibleServiceIds,
-            uint256[] memory eligibleServiceRewards, , ) = _calculateStakingRewards();
+            uint256[] memory eligibleServiceRewards, , , ) = _calculateStakingRewards();
 
         // If there are eligible services, proceed with staking calculation and update rewards for the service Id
         for (uint256 i = 0; i < numServices; ++i) {
             // Get the service index in the eligible service set and calculate its latest reward
             if (eligibleServiceIds[i] == serviceId) {
                 // If total allocated rewards are not enough, adjust the reward value
                 if (totalRewards > lastAvailableRewards) {
-                    reward += (eligibleServiceRewards[i] * lastAvailableRewards) / totalRewards;
+                    reward = (eligibleServiceRewards[i] * lastAvailableRewards) / totalRewards;
                 } else {
-                    reward += eligibleServiceRewards[i];
+                    reward = eligibleServiceRewards[i];
```

Comment:
why the change?

Reply:
In order to separate reward calculation for the current checkpoint and the accumulated one - see above.